bound/kt/src/main/kotlin/web5/sdk/vc/pex/PresentationDefinition.kt

```diff
@@ -1,49 +1,54 @@
 package web5.sdk.vc.pex
 
-import web5.sdk.rust.ConstraintsData as RustCoreConstraintsData
-import web5.sdk.rust.FieldData as RustCoreFieldData
-import web5.sdk.rust.FilterData as RustCoreFilterData
-import web5.sdk.rust.Optionality as RustCoreOptionality
-import web5.sdk.rust.InputDescriptorData as RustCoreInputDescriptorData
-
+import com.fasterxml.jackson.annotation.JsonProperty
+import web5.sdk.Json
 import web5.sdk.rust.PresentationDefinition as RustCorePresentationDefinition
-import web5.sdk.rust.PresentationDefinitionData as RustCorePresentationDefinitionData
-
-typealias ConstraintsData = RustCoreConstraintsData
-typealias FieldData = RustCoreFieldData
-typealias FilterData = RustCoreFilterData
-typealias Optionality = RustCoreOptionality
-typealias InputDescriptor = RustCoreInputDescriptorData
-
-class PresentationDefinition {
-    val id: String
-    val name: String?
-    val purpose: String?
-    val inputDescriptors: List<InputDescriptor>
-
-    internal val rustCorePresentationDefinition: RustCorePresentationDefinition
 
-    constructor(id: String, name: String? = null, purpose: String? = null, inputDescriptors: List<InputDescriptor>) {
-        this.id = id
-        this.name = name
-        this.purpose = purpose
-        this.inputDescriptors = inputDescriptors
-
-        this.rustCorePresentationDefinition = RustCorePresentationDefinition(
-            RustCorePresentationDefinitionData(id, name, purpose, inputDescriptors)
-        )
-    }
-
-    constructor(rustCorePresentationDefinitionData: RustCorePresentationDefinitionData) {
-        this.id = rustCorePresentationDefinitionData.id
-        this.name = rustCorePresentationDefinitionData.name
-        this.purpose = rustCorePresentationDefinitionData.purpose
-        this.inputDescriptors = rustCorePresentationDefinitionData.inputDescriptors
-
-        this.rustCorePresentationDefinition = RustCorePresentationDefinition(rustCorePresentationDefinitionData)
-    }
+data class PresentationDefinition(
+    val id: String,
+    val name: String?,
+    val purpose: String?,
+    @JsonProperty("input_descriptors")
+    val inputDescriptors: List<InputDescriptor>
+) {
+    internal val rustCorePresentationDefinition = RustCorePresentationDefinition(
+        Json.stringify(this)
+    )
 
     fun selectCredentials(vcJwts: List<String>): List<String> {
         return this.rustCorePresentationDefinition.selectCredentials(vcJwts)
     }
-}
\ No newline at end of file
+}
+
+data class InputDescriptor(
```

Comment:
yea I was going back and fourth with this, but seems now we don't have to have toRustCore / fromJustCore now? 

We could `typealias InputDescriptor = RustCoreInputDescriptorData` instead, but It is definitely cleaner (especially if users are going to be creating these types which they wyill) to redefine it like you did here.

So overall, are we back on the train of not doing typealias and just redefining most things since things got updated?

Reply:
we can typealias types which don't require json serialization over the FFI; right now the constraints which lead us to json serialize over the FFI are:

- recursive types (types which have a data member which is itself the same type)
- types which have a "JSON Object" data members

neither of the above two conditions are supported at the FFI layer so we have to json serialize over the boundary, and so because of that, we have to re-define the types at the kt layer